src/ComponentRouter.js

```diff
@@ -42,14 +42,16 @@ const ComponentRouter = React.createClass({
 
 
   componentDidMount() {
-    this.unsubscribe = Store.addThrottledChangeListener(this.onChange, 50);
+    this.unsubscribe =
+      Store.addThrottledChangeListener(this.onChange, 50);
     this.checkDefaultParam(this.props);
   },
 
 
   componentWillUnmount() {
     this.unsubscribe();
-    setTimeout(() => ActionCreator.removeParam({namespace: this.props.namespace}), 0);
+    setTimeout(() => Store.dispatch(ActionCreator.removeParam({
```

Comment:
You don't need this timeout anymore. It was a hack to avoid `dispatch in the middle of dispatch` error.